### PR TITLE
enable csrf in application-connector

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Changes proposed in this pull request:

- Enable CSRF support in application-gateway
- Update documentation

**Related issue(s)**
See also #2852 #3179 #2998 #2996 #3265

### CODE REVIEW

File: docs/application-connector/docs/08-04-register-secured-api.md [hunk 1]
```diff
@@ -104,3 +106,29 @@ kubectl -n kyma-integration patch secrets app-{APP_NAME}-{SERVICE_ID} --patch 'd
   crt: {BASE64_ENCODED_CRT}
   key: {BASE64_ENCODED_KEY}'
 ```
+
+## Register a CSRF-protected API
+
+The application-registry supports CSRF tokens as an additional layer of authorization. To register a CSRF-protected API, you must add the `credentials.csrfInfo` object to the `api` section of the service registration request body.
```
Comment (Tomasz-Smelcerz-SAP) [inline docs/application-connector/docs/08-04-register-secured-api.md]:
```suggestion
The application-registry supports CSRF tokens as an additional layer of API protection. To register a CSRF-protected API, you must add the `credentials.csrfInfo` object to the `api` section of the service registration request body.
```

File: docs/application-connector/docs/08-04-register-secured-api.md [hunk 2]
```diff
@@ -104,3 +106,29 @@ kubectl -n kyma-integration patch secrets app-{APP_NAME}-{SERVICE_ID} --patch 'd
   crt: {BASE64_ENCODED_CRT}
   key: {BASE64_ENCODED_KEY}'
 ```
+
+## Register a CSRF-protected API
+
+The application-registry supports CSRF tokens as an additional layer of API protection. To register a CSRF-protected API, you must add the `credentials.{AUTHENTICATION_METHOD}.csrfInfo` object to the `api` section of the service registration request body.
```
Comment (mmitoraj) [inline docs/application-connector/docs/08-04-register-secured-api.md]:
```suggestion
The Application Registry supports CSRF tokens as an additional layer of API protection. To register a CSRF-protected API, add the `credentials.{AUTHENTICATION_METHOD}.csrfInfo` object to the `api` section of the service registration request body.
```

File: docs/application-connector/docs/08-04-register-secured-api.md [hunk 3]
```diff
@@ -104,3 +106,29 @@ kubectl -n kyma-integration patch secrets app-{APP_NAME}-{SERVICE_ID} --patch 'd
...
+
+Include this field in the service registration request body:
+
+| Field | Description |
+|:-----:|:-----------:|
+| **tokenEndpointURL** | The URL to the upstream service endpoint that exposes CSRF tokens. |
+
+This is an example of the `api` section of the request body for an API secured with both Basic Authentication and a CSRF token.
```
Comment (tomekpapiernik) [inline docs/application-connector/docs/08-04-register-secured-api.md]: I guess the CSFR token verification works as an ADDITIONAL security measure for basic auth- or OAUTH- secured APIs? If so, please make this clear in the introductory paragraph.

Comment (tomekpapiernik) [reply docs/application-connector/docs/08-04-register-secured-api.md]: Nevermind

**Pull request merged**